### PR TITLE
Feature selection by Moran'I score 

### LINKED ISSUE #124: Feature selection by Moran's I
Selecting spatially variable genes based on Moran's I 

link: https://squidpy.readthedocs.io/en/stable/api/squidpy.gr.spatial_autocorr.html

### PR DESCRIPTION
**Pull request by Qirongmao97**
Updates:
       -  Add argprase of utilizing output from squidpy.gr.spatial_neighbors() 
       -  Remove neighborhood calculation in the script
       -  Change branch name

### CODE REVIEW

File: preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    "-m", "--matrix", help="Path to (transformed) counts (as mtx).", required=True
+)
+parser.add_argument(
+    "-f", "--features", help="Path to features (as tsv).", required=True
+)
+parser.add_argument(
+    "-sc", "--spatial_connectivities", help="Path to spatial connectivities (as mtx).", required=True
+)
+parser.add_argument(
+    "-sd", "--spatial_distance", help="Path to spatial distance (as mtx).", required=True
```
Comment (niklasmueboe) [inline preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py:24]: Is the spatial distance needed?

File: preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+
+args = parser.parse_args()
+
+from pathlib import Path
+import pandas as pd
+
+out_dir = Path(args.out_dir)
+
+# Output files 
+feature_selection_file = out_dir / "features_MoranI.tsv"
```
Comment (niklasmueboe) [inline preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py:48]: rename output to features.tsv to be consistent with current setup

File: preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+features_df = adata.var.copy()
+
+## Calculate Moran's I score for each gene
+sq.gr.spatial_autocorr(adata, mode="moran", genes=adata.var_names,show_progress_bar=True)
+
+
+## Selecting top spatially variable genes based on Moran's I score
+SVG = (
+    adata.uns["moranI"]["I"].sort_values(ascending=False).head(n_top_genes).index.tolist()
```
Comment (niklasmueboe) [inline preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py:119]: consider using [nlargest](http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.nlargest.html)

File: preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py
Comment (niklasmueboe) [inline preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py:1]: I think we should rename the file to spatially_variable_genes_moransI

**Review by niklasmueboe — CHANGES_REQUESTED** [review]

File: preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py (hunk 1 above)
Comment (Qirongmao97) [reply preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py]: No, spatial distance has deleted.

File: preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py (hunk 2 above)
Comment (Qirongmao97) [reply preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py]: Done

File: preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py (hunk 3 above)
Comment (Qirongmao97) [reply preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py]: Done

File: preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py
Comment (Qirongmao97) [reply preprocessing/feature_selection_MoranI/spatially_variable_genes_scanpy.py:1]: The file name has changed (both .py and .yml）

**Review by niklasmueboe — APPROVED** [review]

File: preprocessing/feature_selection_MoranI/spatially_variable_genes_moransI.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    "-c", "--coordinates", help="Path to coordinates (as tsv).", required=True
+)
+parser.add_argument(
+    "-m", "--matrix", help="Path to (transformed) counts (as mtx).", required=True
+)
+parser.add_argument(
+    "-f", "--features", help="Path to features (as tsv).", required=True
+)
+parser.add_argument(
+    "-sc", "--spatial_connectivities", help="Path to spatial connectivities (as mtx).", required=True
```
Comment (niklasmueboe) [inline preprocessing/feature_selection_MoranI/spatially_variable_genes_moransI.py:21]: rename the arg to be consistent with prior nomenclature

`parser.add_argument(
    "-n",
    "--neighbors",
    help="Path to neighbor definitions. Square matrix (not necessarily symmetric) where each row contains the neighbors of this observation (as mtx).",
    required=False
)`

**Review by niklasmueboe — CHANGES_REQUESTED** [review]

**Review by niklasmueboe — APPROVED** [review]

**Pull request merged**